### PR TITLE
Update Roslyn.Services.UnitTests.Utilities to target netcoreapp3.1

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -102,7 +102,7 @@
     <MicrosoftVisualStudioCodeAnalysisSdkUIVersion>15.8.27812-alpha</MicrosoftVisualStudioCodeAnalysisSdkUIVersion>
     <MicrosoftVisualStudioCodingConventionsVersion>1.1.20180503.2</MicrosoftVisualStudioCodingConventionsVersion>
     <MicrosoftVisualStudioComponentModelHostVersion>16.0.467</MicrosoftVisualStudioComponentModelHostVersion>
-    <MicrosoftVisualStudioCompositionVersion>15.6.36</MicrosoftVisualStudioCompositionVersion>
+    <MicrosoftVisualStudioCompositionVersion>16.4.11</MicrosoftVisualStudioCompositionVersion>
```
Comment (sharwell) [inline eng/Versions.props:105]: 📝 This causes test failures, which are getting fixed by https://github.com/dotnet/roslyn-sdk/pull/562

File: src/Workspaces/CoreTestUtilities/OptionsCollection.cs [hunk 2]
```diff
@@ -4,12 +4,15 @@
...
 
-using System;
 using System.Collections;
 using System.Collections.Generic;
 using Microsoft.CodeAnalysis.CodeStyle;
 using Microsoft.CodeAnalysis.Options;
+
+#if !NETCOREAPP
+using System;
 using Roslyn.Utilities;
```
Comment (genlu) [inline src/Workspaces/CoreTestUtilities/OptionsCollection.cs:14]: Why this #if is needed?

Comment (sharwell) [reply src/Workspaces/CoreTestUtilities/OptionsCollection.cs:14]: The using directives are only used in one framework.

Comment (jaredpar) [reply src/Workspaces/CoreTestUtilities/OptionsCollection.cs:14]: Minor curiousity: what about `Roslyn.Utilities` gets used in .NET Core here? There aren't any particular code changes in the PR that hook into new APIs.

Comment (sharwell) [reply src/Workspaces/CoreTestUtilities/OptionsCollection.cs:14]: @jaredpar None of them do, hence the exclusion for that build 😄

Comment (jaredpar) [reply src/Workspaces/CoreTestUtilities/OptionsCollection.cs:14]: 🤦 

This is what I get for commenting before finishing my coffee

**Pull request merged**